x-pack/plugin/core/src/main/java/org/elasticsearch/license/internal/TrialLicenseVersion.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.license.internal;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Sometimes we release a version with a bunch of cool new features, and we want people to be able to start a new trial license in a cluster
+ * that's already used a trial and let it expire. This class controls when we do that. The serialization of this class is designed to
+ * maintain compatibility with old-school Elasticsearch versions (specifically the {@link org.elasticsearch.Version} class).
+ */
+public class TrialLicenseVersion implements ToXContentFragment, Writeable {
+
+    // Increment this when we want users to be able to start a new trial. Note that BWC with old versions of Elasticsearch cause this to be
+    // limited to a maximum of 99 (inclusive). See the Version class for more info.
+    public static final TrialLicenseVersion CURRENT = new TrialLicenseVersion(8);
+
+    private final int era;
+
+    public TrialLicenseVersion(int era) {
+        this.era = era;
+    }
+
+    public TrialLicenseVersion(StreamInput in) throws IOException {
+        this.era = (in.readVInt() - 99) / 1000000; // Copied from the constructor of Version
```

Comment:
In other new *Version forks from Version we have used the exact same serialization but diverging going forward (a bigger integer). Rather than try to reduce this down to the major version number from previous Version instances, can we pin the latest version id as the "current", and not do any manipulation to the id when read?

Reply:
Thanks, that is a better way of doing it - frankly this is the first thing that occurred to me and upon reflection that's obviously a better way of doing it. I'll make the updates.